azure-spring-boot/src/main/java/com/microsoft/azure/keyvault/spring/KeyVaultOperation.java

```diff
@@ -3,91 +3,148 @@
  * Licensed under the MIT License. See LICENSE in the project root for
  * license information.
  */
-
 package com.microsoft.azure.keyvault.spring;
 
+import com.azure.core.http.rest.PagedIterable;
 import com.azure.security.keyvault.secrets.SecretClient;
 import com.azure.security.keyvault.secrets.models.KeyVaultSecret;
+import com.azure.security.keyvault.secrets.models.SecretProperties;
 import lombok.extern.slf4j.Slf4j;
 import org.springframework.lang.NonNull;
-import org.springframework.util.Assert;
-import org.springframework.util.StringUtils;
 
-import java.util.ArrayList;
-import java.util.Collection;
+import java.util.LinkedHashMap;
 import java.util.List;
 import java.util.Locale;
-import java.util.Optional;
-import java.util.stream.Collectors;
+import java.util.Timer;
+import java.util.TimerTask;
 import java.util.stream.Stream;
 
 @Slf4j
 public class KeyVaultOperation {
-    
+
     /**
      * Stores the case sensitive flag.
      */
     private final boolean caseSensitive;
 
-    private final SecretClient keyVaultClient;
-    private final String vaultUri;
-    private volatile List<String> secretNames;
-    private final boolean secretNamesAlreadyConfigured;
-    private final long secretNamesRefreshIntervalInMs;
-    private volatile long secretNamesLastUpdateTime;
+    /**
+     * Stores the properties.
+     */
+    private LinkedHashMap<String, String> properties = new LinkedHashMap<>();
+
+    /**
+     * Stores the secret client.
+     */
+    private final SecretClient secretClient;
+
+    /**
+     * Stores the secret keys.
+     */
+    private List<String> secretKeys;
```

Comment:
To make code less confusing, I suggest to change variable/method name:

* secretKeys -> configuredSecretKeys
* properties  -> keyVaultSecretCache
* refreshProperties -> refreshkeyVaultSecretCache

Reply:
OK.